pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/access"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/label"
+	dockerutil "github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/log"
+	v1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/status"
+	pkgsync "github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+type Deployer struct {
+	accessor access.Accessor
+	debugger debug.Debugger
+	logger   log.Logger
+	monitor  status.Monitor
+	syncer   pkgsync.Syncer
+
+	cfg                *v1.DockerDeploy
+	client             dockerutil.LocalDaemon
+	deployedContainers map[string]string // imageName -> containerID
+	network            string
+	once               sync.Once
+}
+
+func NewDeployer(cfg dockerutil.Config, labeller *label.DefaultLabeller, d *v1.DockerDeploy, resources []*v1.PortForwardResource) (*Deployer, error) {
+	client, err := dockerutil.NewAPIClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Deployer{
+		cfg:                d,
+		client:             client,
+		deployedContainers: make(map[string]string),
+		network:            fmt.Sprintf("skaffold-network-%s", uuid.New().String()),
+		// TODO(nkubala): implement components
+		accessor: &access.NoopAccessor{},
+		debugger: &debug.NoopDebugger{},
+		logger:   &log.NoopLogger{},
+		monitor:  &status.NoopMonitor{},
+		syncer:   &pkgsync.NoopSyncer{},
+	}, nil
+}
+
+func (d *Deployer) TrackBuildArtifacts(artifacts []graph.Artifact) {
+	// TODO(nkubala): implement with components
+}
+
+func (d *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Artifact) error {
+	var err error
+	d.once.Do(func() {
+		err = d.client.NetworkCreate(ctx, d.network)
+	})
+	if err != nil {
+		return fmt.Errorf("creating skaffold network %s: %w", d.network, err)
+	}
+	for _, b := range builds {
+		// TODO(nkubala): parallelize this
+		if !util.StrSliceContains(d.cfg.Images, b.ImageName) {
```

Comment:
It seems like a reasonable request to be able to deploy an existing image?  E.g., busybox, or mongodb?